concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,571 @@
+-- Elective stay: If there is no information on surgery in an elective stay, we assumed all cases to be -> "no elective surgery"
+-- There are a lot of missing values, especially for urine output. Hence, we have created 3 OASIS summary scores:
+-- 1) No imputation, values as is with missings. 2) Imputation by assuming a Best Case scenario. 3) Imputation by assuming a Worst Case scenario. 
+
+-- Note:
+--  The score is calculated for *all* ICU patients, with the assumption that the user will subselect appropriate patientunitstayid.
+
+DROP TABLE IF EXISTS `physionet-data.eicu_crd_derived.pivoted_OASIS` ;
+CREATE TABLE `physionet-data.eicu_crd_derived.pivoted_OASIS` AS
+
+WITH 
+
+-- Pre-ICU stay LOS -> directly convert from minutes to hours
+pre_ICU_LOS_OASIS AS (
+SELECT patientunitstayid AS pid_LOS
+  ,CASE
+      WHEN hospitaladmitoffset < (0.17*60) THEN 5
+      WHEN hospitaladmitoffset BETWEEN (0.17*60) AND (4.94*60) THEN 3
+      WHEN hospitaladmitoffset BETWEEN (4.94*60) AND (24*60) THEN 0
+      WHEN hospitaladmitoffset BETWEEN (24.01*60) AND (311.80*60) THEN 2
+      WHEN hospitaladmitoffset > (311.80*60)  THEN 1
+      ELSE NULL
+      END AS pre_ICU_LOS_OASIS
+    FROM `physionet-data.eicu_crd.patient`
+    WHERE hospitaladmitoffset IS NOT NULL
+     AND patientunitstayid IS NOT NULL
+    GROUP BY pid_LOS, hospitaladmitoffset
+)
+
+-- Age 
+-- Change age from string to integer
+, age_numeric AS (
+  
+  SELECT patientunitstayid 
+  , CASE
+    WHEN age = "> 89" THEN 91
+    ELSE SAFE_CAST(age AS INT64) 
+    END AS age_num
+    FROM `physionet-data.eicu_crd.patient`
+    WHERE age IS NOT NULL
+      AND patientunitstayid IS NOT NULL
+    GROUP BY patientunitstayid, age
+)
+
+-- Get the information itself in a second step
+, age_OASIS AS (
+    SELECT patientunitstayid AS pid_age
+    , CASE
+    WHEN MAX(age_num < 24) THEN 0
+    WHEN MAX(age_num) BETWEEN 24 AND 53 THEN 3
+    WHEN MAX(age_num) BETWEEN 54 AND 77 THEN 6
+    WHEN MAX(age_num) BETWEEN 78 AND 89 THEN 9
+    WHEN MAX(age_num) > 89 THEN 7
+    ELSE NULL
+    END AS age_OASIS
+    FROM age_numeric
+    GROUP BY pid_age, age_num
+)
+
+-- GCS, Glasgow Coma Scale
+-- Merge information from two tables into one
+, merged_GCS AS (
+    
+    SELECT pivoted_gcs.patientunitstayid, pivoted_gcs.gcs
+    FROM `physionet-data.eicu_crd_derived.pivoted_gcs` AS pivoted_gcs
+   
+    FULL OUTER JOIN(
+      SELECT patientunitstayid, MIN(CAST(physicalexamvalue AS NUMERIC)) AS gcs
+      FROM `physionet-data.eicu_crd.physicalexam`
+      WHERE  (
+      (physicalExamPath LIKE "notes/Progress Notes/Physical Exam/Physical Exam/Neurologic/GCS/_" OR
+       physicalExamPath LIKE "notes/Progress Notes/Physical Exam/Physical Exam/Neurologic/GCS/__")
+      AND (physicalexamoffset > 0 AND physicalexamoffset <= 1440 ) -- consider only first 24h
+      AND physicalexamvalue IS NOT NULL
+      AND patientunitstayid IS NOT NULL
+      )
+  GROUP BY patientunitstayid
+  )
+  AS physicalexam
```

Comment:
It's stylistic, but I would avoid mixing in-line views (`WITH alias AS ...`) with queries within FROM / JOIN clauses (`FROM ( ... ) AS alias`). I think it's easier to read if you stick with the former (`, physicalexam AS `).